hydroflow_lang/src/graph/ops/fold.rs

```diff
@@ -90,9 +90,14 @@ pub const FOLD: OperatorConstraints = OperatorConstraints {
                         #[allow(clippy::redundant_closure_call)]
                         let mut #accumulator_ident = (#initializer_func_ident)();
 
+                        #[inline(always)]
```

Comment:
@MingweiSamuel @zzlk this helps guide the type inference for the accumulator function; otherwise you can get code that typechecks at the staging level but fails to infer the type for the `&mut A` parameter of the closure when the final code is generated

should also help with ergonomics in general by eliminating the need to manually annotate that parameter

Reply:
does this issue also come up with reduces and several other places? I think we can also address those as we go. I am in favor of getting this in and getting some use out of it to find out where the pain points are.